ace-ecosystem/ace2-ams
# Filter Chips & Test Warnings Fixes

Filter Chips:

This PR adds simple 'filter chips' and associated toolbar to display currently applied filters without needing to open the filter modal. 

In order to accomodate the new filter chip container, some of the filter actions have been moved around, I anticpate the UI of the filter toolbar will continue to change.

Also, some of the filterOptions properties have changed to be more abstract, and more easily used outside of services code.

Test Warnings Fixes: 

As an add-on to this PR, I fixed some of the warnings that showed up when running unit tests. Some reflected actual bugs in the code, some were due to errors in tests themselves. Either way, there should be no more warnings and test output should be cleaner.

Detailed changelist:
- frontend/src/components/Alerts/AnalyzeAlertForm.vue
	- Properly initialize alertDate
- frontend/src/components/Filters/FilterChip.vue
	- Renamed from 'FilterTag'
	- Add new filter chip
	- Eventually, can support multiple values per given filter type
- frontend/src/components/Filters/FilterChipContainer.vue
	- Renamed from 'FilterTagContainer'
	- Add new filter chip container to display chips for set filters of given filterType
- frontend/src/components/Filters/FilterInput.vue
	- Moved file to /filters
	- Update for changed property names
- frontend/src/components/Filters/TheFilterToolbar.vue
	- Rearranged buttons
	- Added toolbar for filter chips
	- Add data-cy properties for better E2E testing
- frontend/src/components/Modals/FilterModal.vue
	- Update import for moved file
- frontend/src/components/Node/NodeTag.vue
	- Change NodeTag to be based off Tag component over Chip
		- Easier to differentiate from filter chips, also looks more like a 'tag' (naturally)
- frontend/src/components/UserInterface/DateRangePicker.vue
	- Add data-cy properties for better E2E testing
- frontend/src/etc/constants.ts
	- Update alertFilters to use new filterOption properties
- frontend/src/etc/helpers.ts
	- Add isObject method
	- Add typing
- frontend/src/models/base.ts
	- Change filterOption properties
- frontend/src/pages/Alerts/ManageAlerts.vue
	- Update for modified filterOption properties
- frontend/src/services/api/alert.ts
	- Update for modified filterOption properties
- frontend/src/stores/filter.ts
	- Prevent addition of empty/null filter values
